Python/ceval.c

```diff
@@ -4066,7 +4066,88 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
             CHECK_EVAL_BREAKER();
             DISPATCH();
         }
+        case TARGET(CALL_FUNCTION_ADAPTIVE): {
+            SpecializedCacheEntry *cache = GET_CACHE();
+            if (cache->adaptive.counter == 0) {
+                next_instr--;
+                if (_Py_Specialize_CallFunction(BUILTINS(), stack_pointer,
+                    cache->adaptive.original_oparg, next_instr, cache) < 0) {
+                    goto error;
+                }
+                DISPATCH();
+            }
+            else {
+                STAT_INC(CALL_FUNCTION, deferred);
+                cache->adaptive.counter--;
+                oparg = cache->adaptive.original_oparg;
+                JUMP_TO_INSTRUCTION(CALL_FUNCTION);
+            }
+        }
+        case TARGET(CALL_CFUNCTION_FAST): {
+            DEOPT_IF(cframe.use_tracing, CALL_FUNCTION);
+            /* Builtin METH_FASTCALL functions, without keywords */
+            SpecializedCacheEntry *caches = GET_CACHE();
+            _PyAdaptiveEntry *cache0 = &caches[0].adaptive;
+            uint8_t nargs = cache0->original_oparg;
+            PyObject **pfunc = &PEEK(nargs + 1);
```

Comment:
Drop `pfunc` and the extra indirection. `PyObject *callable = PEEK(nargs + 1);` is clearer.